rackcity/views/rack_views.py

```diff
@@ -76,7 +76,31 @@ def rack_create(request):
     """
     Create racks within specified range.
     """
-    return JsonResponse({}, status=HTTPStatus.INTERNAL_SERVER_ERROR)
+    range_serializer = RackRangeSerializer(data=request.data)
+    if not range_serializer.is_valid():
+        return JsonResponse(
+            {"failure_message": "Invalid keys or values in request."},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    racks = Rack.objects.filter(
+        rack_num__range=range_serializer.get_number_range(),  # inclusive range
+        row_letter__range=range_serializer.get_row_range(),
+    )
+    if racks.count() > 0:
+        failure_message = "Racks cannot be created because racks within " + \
+            "the range already exist."
+        return JsonResponse(
```

Comment:
Seems like you're following the fail early pattern here which we don't really do elsewhere. I don't dislike it but worth tracking and considering standardizing one way or the other at some point down the line

Reply:
tracking engaged